packages/mobile/src/store/offline-downloads/sagas.ts

```diff
@@ -253,6 +256,32 @@ function* watchAddTrackToPlaylist() {
   )
 }
 
+function* downloadNewFavoriteIfNecessary({
+  trackId
+}: ReturnType<typeof savedPageActions.addLocalSave>) {
+  const areFavoritesDownloaded = yield* select(
```

Comment:
Is there a possible race condition where we haven't hydrated the offline downloads slice before getting here? If so, we can wait for `isDoneLoadingFromDisk` to be true.